service/src/test/java/uk/nhs/adaptors/oneoneone/cda/report/comparator/ResourceComparatorTest.java

```diff
@@ -0,0 +1,77 @@
+package uk.nhs.adaptors.oneoneone.cda.report.comparator;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+import java.util.Date;
+
+import org.hl7.fhir.dstu3.model.ReferralRequest;
+import org.hl7.fhir.dstu3.model.ResourceType;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class ResourceComparatorTest {
+    @InjectMocks
+    private ResourceComparator resourceComparator;
+    @Mock
+    private ReferralRequest referralRequest1;
+    @Mock
+    private ReferralRequest referralRequest2;
+
+    @Test
+    public void shouldReturnNegativeOneForValidDates() {
```

Comment:
All these tests check comparator functionality pretty well, however they're hard to understand without digging into the code.

What I'd do:

    @Test
    public void shouldReturnOneWhenFirstResourceIsLater() {
        ReferralRequest first = new ReferralRequest().setAuthoredOn(LATER_DATE);
        ReferralRequest second = new ReferralRequest().setAuthoredOn(EARLIER_DATE);

        assertThat(resourceComparator.compare(first, second)).isEqualTo(1);
    }